subcommand/health-sync/command.go

```diff
@@ -0,0 +1,183 @@
+	if err != nil {
+		return fmt.Errorf("constructing consul client: %s", err)
+	}
+	c.consulClient = consulClient
+
+	taskMeta, err := awsutil.ECSTaskMetadata()
+	if err != nil {
+		return err
+	}
+
+	// Duplicate code from mesh-init. Service ID must match here and in mesh-init.
+	serviceName := taskMeta.Family
+	serviceID := fmt.Sprintf("%s-%s", serviceName, taskMeta.TaskID())
+	c.log.Info("discovered service", "serviceID", serviceID)
+
+	currentStatuses := map[string]string{}
+	parsedContainerNames := strings.Split(c.flagContainerNames, ",")
+
+	for {
+		currentStatuses = c.syncChecks(currentStatuses, serviceID, parsedContainerNames)
+	}
+}
+
+func (c *Command) ignoreSIGTERM() {
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGTERM)
+	// Don't need to do anything for now. Just catch the SIGTERM so we don't exit.
+	// And, print when we receive the SIGTERM
+	go func() {
+		for sig := range sigs {
+			c.log.Info("signal received", "signal", sig)
+		}
+	}()
+}
+
+func (c *Command) syncChecks(currentStatuses map[string]string, serviceID string, parsedContainerNames []string) map[string]string {
+	taskMeta, err := awsutil.ECSTaskMetadata()
+	if err != nil {
+		c.log.Error(err.Error())
+	}
+	containersToSync, missingContainers := findContainersToSync(parsedContainerNames, taskMeta)
+	for _, name := range missingContainers {
+		c.UI.Error(fmt.Sprintf("container %s not found in task metadata", name))
+	}
+
+	for _, container := range containersToSync {
+		c.log.Info("Updating Consul TTL check from ECS container health",
+			"name", container.Name,
+			"status", container.Health.Status,
+			"statusSince", container.Health.StatusSince,
+			"exitCode", container.Health.ExitCode,
+		)
+
+		previousStatus := currentStatuses[container.Name]
+		if container.Health.Status != previousStatus {
+			err = c.updateConsulHealthStatus(serviceID, container)
+			if err != nil {
+				c.UI.Error(fmt.Sprintf("failed to update Consul health status: %s", err.Error()))
+			} else {
+				c.log.Info(fmt.Sprintf("Container %s health check updated in Consul", container.Name))
+				currentStatuses[container.Name] = container.Health.Status
+			}
+		}
+
+		// The rate limit is about 40 per second, so 1 second pulling seems reasonable
+		time.Sleep(1 * time.Second)
+	}
+
+	return currentStatuses
+}
+
+func findContainersToSync(containerNames []string, taskMeta awsutil.ECSTaskMeta) ([]*awsutil.ECSTaskMetaContainer, []string) {
+	ecsContainers := []*awsutil.ECSTaskMetaContainer{}
+	missing := []string{}
+
+ContainerNames:
+	for _, container := range containerNames {
+		for _, ecsContainer := range taskMeta.Containers {
+			if ecsContainer.Name == container {
+				ecsContainers = append(ecsContainers, &ecsContainer)
+				continue ContainerNames
+			}
+		}
+		missing = append(missing, container)
+	}
+	return ecsContainers, missing
+}
+
+func ecsHealthToConsulHealth(ecsHealth string) string {
+	if ecsHealth != "HEALTHY" {
```

Comment:
nit: a comment here that lists the ECS health statuses would be good (`UNKNOWN`, `HEALTHY`, or `UNHEALTHY`) so it's clearer why it's a binary choice of either critical or passing for the Consul check.